Products/CMFPlone/browser/recyclebin.py

```diff
@@ -0,0 +1,592 @@
+    def handle_delete(self, action):
+        """Delete selected items handler"""
+        data, errors = self.extractData()
+
+        # Get the selected items from the request directly
+        selected_items = self.request.form.get("selected_items", [])
+        if not isinstance(selected_items, list):
+            selected_items = [selected_items]
+
+        if not selected_items:
+            IStatusMessage(self.request).addStatusMessage(
+                "No items selected for deletion.", type="info"
+            )
+            return
+
+        recycle_bin = getUtility(IRecycleBin)
+        deleted_count = 0
+        for item_id in selected_items:
+            if recycle_bin.purge_item(item_id):
+                deleted_count += 1
+
+        message = f"{deleted_count} item{'s' if deleted_count != 1 else ''} permanently deleted."
+        IStatusMessage(self.request).addStatusMessage(message, type="info")
+
+    @button.buttonAndHandler("Empty Recycle Bin", name="empty")
+    def handle_empty(self, action):
+        """Empty recycle bin handler"""
+        data, errors = self.extractData()
+
+        recycle_bin = getUtility(IRecycleBin)
+        items = recycle_bin.get_items()
+        deleted_count = 0
+
+        for item in items:
+            item_id = item["recycle_id"]
+            if recycle_bin.purge_item(item_id):
+                deleted_count += 1
+
+        message = f"Recycle bin emptied. {deleted_count} item{'s' if deleted_count != 1 else ''} permanently deleted."
+        IStatusMessage(self.request).addStatusMessage(message, type="info")
+
+
+class RecycleBinView(BrowserView):
+    """Browser view for recycle bin management"""
+
+    template = ViewPageTemplateFile("templates/recyclebin.pt")
+
+    def __call__(self):
+        # Initialize form
+        form = RecycleBinForm(self.context, self.request)
+        form.update()
+
+        # Check if form was submitted and return template
+        return self.template()
+
+    def get_recycle_bin(self):
+        """Get the recycle bin utility"""
+        return getUtility(IRecycleBin)
+
+    def get_search_query(self):
+        """Get the search query from the request"""
+        return self.request.form.get("search_query", "")
+
+    def get_sort_option(self):
+        """Get the current sort option from the request"""
+        return self.request.form.get("sort_by", "date_desc")
+
+    def get_filter_type(self):
+        """Get the content type filter from the request"""
+        return self.request.form.get("filter_type", "")
+
+    def get_sort_labels(self):
+        """Get a dictionary of human-readable sort option labels"""
+        return {
+            "date_desc": "Newest First (default)",
+            "date_asc": "Oldest First",
+            "title_asc": "Title (A-Z)",
+            "title_desc": "Title (Z-A)",
+            "type_asc": "Type (A-Z)",
+            "type_desc": "Type (Z-A)",
+            "path_asc": "Path (A-Z)",
+            "path_desc": "Path (Z-A)",
```

Comment:
For all these sorts, let's make sure that they sort alphanumerically, not by bytes. Some sorts may use the ASCII value of the character, which would be unexpected and annoying. For example, in ascending order should be the following.

- A title
- b title
- C title